components/sup/src/manager/service/mod.rs

```diff
@@ -1280,8 +1280,27 @@ mod tests {
         let fscfg = FsCfg::new("/tmp");
         let afs = Arc::new(fscfg);
         let gs = Arc::new(RwLock::new(manager::GatewayState::default()));
-        let service = Service::new(asys, install, spec, afs, Some("haha"), gs)
-            .expect("I wanted a service to load, but it didn't");
+
+        Service::new(asys, install, spec, afs, Some("haha"), gs)
+            .expect("I wanted a service to load, but it didn't")
+    }
+
+    #[test]
+    fn health_check_is_due_at_creation() {
+        let service = initialize_test_service();
+        assert!(
+            service.scheduled_health_check.is_some(),
+            "Expected a scheduled health check"
+        );
+        assert!(
+            service.scheduled_health_check.unwrap() < Instant::now(),
+            "Expected health check due at creation"
+        );
```

Comment:
This works fine, but just a rust testing FYI, you could structure it like this for a bit less redundancy
```rust
if let Some(scheduled_health_check) = service.scheduled_health_check {
    assert!(scheduled_health_check < Instant::now(), "Expected health check due at creation");
} else {
    panic("Expected a scheduled health check");
}
```